src/Services/Settings.js

```diff
@@ -85,11 +87,45 @@ class Settings {
     // Ensure new settings variables get added by merging all the settings,
     // where the current values take precedence. 
     let currentSettings = this._getAllRaw()
+
+    // Add any non-additive settings changes here by creating a function which
+    // handles the settings change in question.
+    currentSettings = this.migrateMnemonicSettings(currentSettings);
+
     currentSettings = _.merge({}, initialSettings, currentSettings);
 
     // Apply the merged settings
     this.setAll(currentSettings);
   }
+
+  migrateMnemonicSettings(currentSettings) {
+    // If we're migrating a settings file from before we used a persistent,
+    // randomly generated mnemonic by default, randomizeMnemonic on start will
+    // be undefined.
+    if (currentSettings.server.randomizeMnemonicOnStart === undefined) {
+
+      // Before we added the randomizeMnemonicOnStart flag, the absence of a
+      // mnemonic meant that we wanted a random one one each run. We want to
+      // preserve this preference.
+      if (currentSettings.server.mnemonic === "") {
+        currentSettings.server.randomizeMnemonicOnStart = true;
+      } else if (currentSettings.server.mnemonic === oldDefaultMnemonic || !currentSettings.server.mnemonic) {
+
+        // This will cause a new mnemonic to be generated and persisted only in
+        // the case when the old default mnemonic was being used.
+        currentSettings.server.mnemonic = null;
+      }
+    }
+
+    return currentSettings;
+  }
+
+  _onNewMnemonic(mnemonic) {
```

Comment:
Should we just rename this function to `setMnemonic()` since you're calling it from the outside [here](https://github.com/trufflesuite/ganache/pull/378/files#diff-1c90ff38a08209f9ebd4d05d1e43358eR114)?

Reply:
Good question. I struggled with a name for this, and I still don't like the one I have. That said, I don't like `setMnemonic` much either. I started with it, but I changed it because setters should generally be dumb and either set the thing you want them to set, or throw a validation error. Since this function might decide to not modify state without throwing, I didn't feel like calling it `set...` was a good fit.

I'm open to other ideas, though. Since it's really an event handler, I went with the `on...` naming. Though it's called externally, so I should probably drop the underscore